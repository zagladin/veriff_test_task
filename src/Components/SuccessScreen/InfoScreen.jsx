import React from 'react';
import PropTypes from 'prop-types';
import styles from './InfoScreen.module.scss';

import Button from '../Button/Button';

const InfoScreen = (
  {
    icon,
    text,
    clickCallback,
    buttonText,
  },
) => (
  <div className={styles.infoScreenWrapper}>
    <div>{icon}</div>
    <div>{text}</div>
    {clickCallback && (
    <Button
      disabled={false}
      onClick={clickCallback}
    >
      {buttonText}
    </Button>
    )}
  </div>
);

InfoScreen.propTypes = {
  icon: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  clickCallback: PropTypes.func.isRequired,
  buttonText: PropTypes.string.isRequired,
};

export default React.memo(InfoScreen);
