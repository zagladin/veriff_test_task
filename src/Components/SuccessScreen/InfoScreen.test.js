import React from 'react';
import renderer from 'react-test-renderer';
import InfoScreen from './InfoScreen';

const mockData = {
  clickCallback: () => console.log('Click'),
  text: 'Info screen text',
  icon: 'ðŸ’€',
  buttonText: 'Click me',
};

it('should match the snapshot', () => {
  const checkSection = renderer
      .create(
          <InfoScreen
              text={mockData.text}
              buttonText={mockData.buttonText}
              icon={mockData.icon}
              clickCallback={mockData.clickCallback}/>,
      )
      .toJSON();
  expect(checkSection).toMatchSnapshot();
});

