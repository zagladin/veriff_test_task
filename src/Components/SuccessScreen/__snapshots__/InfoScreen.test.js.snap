// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="infoScreenWrapper"
>
  <div>
    ðŸ’€
  </div>
  <div>
    Info screen text
  </div>
  <button
    className="Button"
    disabled={false}
    onClick={[Function]}
  >
    Click me
  </button>
</div>
`;
